zeebe/exporters/elasticsearch-exporter/src/test/java/io/camunda/zeebe/exporter/ElasticsearchExporterIT.java

```diff
@@ -240,6 +243,18 @@ private <T extends RecordValue> Record<T> generateRecord(final ValueType valueTy
         valueType, r -> r.withBrokerVersion(VersionUtil.getVersionLowerCase()));
   }
 
+  private <T extends RecordValue> Record<T> generateSupportedRecord(
+      final UnaryOperator<Builder<T>> modifier) {
+    Record<T> record = factory.generateRecord(modifier);
+    while (!TestSupport.provideValueTypes()
```

Comment:
❌ We are `collecting` the set in every iteration again now.

🔧 instead of retrying to generate a record until success, is it an idea to randomly pick a valuetype of this stream and set it in the record?